### PR TITLE
add copy dashboard page feature

### PR DESCRIPTION
**Pull request by gaecoli**
## What type of PR is this? (check all applicable)

- [x] Feature

## Description
add copy dashboard page feature

### CODE REVIEW

**Comment by yankeeguyu** [comment]
In my local machine, this feature can work! I don't know what should i review.

File: client/app/pages/dashboards/hooks/useDuplicateDashboard.js [hunk 1]
```diff
@@ -0,0 +1,13 @@
+import { useCallback } from "react";
+import { Dashboard } from "@/services/dashboard";
+
+export default function useDuplicateDashboard(dashboard) {
+  const duplicateDashboard = useCallback(() => {
+    const dashboardSlug = dashboard.slug;
+    Dashboard.copy({ slug: dashboardSlug }).then(({ slug }) => {
+      window.open(`dashboard/${slug}`);
+    });
```
Comment (kravets-levko) [inline client/app/pages/dashboards/hooks/useDuplicateDashboard.js:9]: Please address this according to my comment in your previous PR: https://github.com/getredash/redash/pull/5232#discussion_r511338830 There is a link to similar function we use for Fork Query - just implement everything in the same way (I also explained why this is needed).

**Review by kravets-levko — CHANGES_REQUESTED** [review]
I'm not sure why you preferred to replace previous PR with this one - they look pretty the same. Anyway, please see my comments

File: client/app/pages/dashboards/hooks/useDuplicateDashboard.js (hunk 1 above)
Comment (yankeeguyu) [reply client/app/pages/dashboards/hooks/useDuplicateDashboard.js]: Hi, i modefied your comment, bug i found that query and dashboard have different structures, so the lack of `geturl` method makes the current dashboard unable to reposition! So maybe someone can help me to check this problem! Thank you!

File: client/app/pages/dashboards/hooks/useDuplicateDashboard.js [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+  const duplicateDashboard = useCallback(() => {
+    const tabName = `duplicatedDashboardTab/${Math.random().toString()}`;
+    // We should open tab here because this moment is a part of user interaction;
+    // later browser will block such attempts
+    const tab = window.open("", tabName);
+
+    setIsDuplicating(true);
+    Dashboard.fork({ slug: dashboard.slug })
+      .then(newDashboard => {
+        tab.location = keepCurrentUrlParams(newDashboard.getUrl(true));
```
Comment (yankeeguyu) [inline client/app/pages/dashboards/hooks/useDuplicateDashboard.js:30]: This `getUrl()` should be rewrite!

**Comment by jhult** [comment]
@gaecoli, any updates on this? It would be a nice feature to have available.

**Comment by gaecoli** [comment]
> @gaecoli, any updates on this? It would be a nice feature to have available.

I will update, but the encoding format should not meet the requirements of the redash community.

**Comment by gaecoli** [comment]
> @gaecoli, any updates on this? It would be a nice feature to have available.

It merged in master! @jhult

**Pull request closed without merging**